server/app/modules/FeatureFlagsModule.java

```diff
@@ -0,0 +1,21 @@
+package modules;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import annotations.FeatureFlags.StatusTrackingEnabled;
+import com.google.inject.AbstractModule;
+import com.google.inject.Provides;
+import com.typesafe.config.Config;
+
+public class FeatureFlagsModule extends AbstractModule {
+  @Override
+  public void configure() {
+  }
+
+  @Provides
+  @StatusTrackingEnabled
+  public boolean provideStatusTrackingEnabled(Config config) {
+    return checkNotNull(config).getBoolean("status_tracking_enabled");
```

Comment:
This works, but is there a need to add Guice annotations when the config is already injected at the callsite? For example, the flag could be a static String and the callsites could evaluate against the config.

One argument in favor of Guice annotations would be it'd be easier to mock out data in unit tests.

Reply:
1. I think it's good to abstract these mechanics out, especially as we add more Feature Flags and ones that will be long lived.  The values will also be treated more "readably" through the code base as class ctors will generally have to accept the injected value and save it as a class field.
2. I hope we can support url overrides in dev/staging and having that handled in one place is good.
3. Not all involved classes will have the config injected also I think.
4. Testing does seem easier and the "concern" is treated as a first party thing as you mention

Reply:
Definitely seems like reasonable rationale. Another nice thing that would fall out of this is if we wanted to eventually allow admins / specific user types to "force" experiments. Having a single Guice-injection site that can wrap that decision up seems preferable to having each consumer of the feature flag have to apply that logic.

Reply:
yeah, def having wide eyed views into that future :D